cmd/snap-repair/runner.go

```diff
@@ -655,14 +656,52 @@ func verifySignatures(a asserts.Assertion, workBS asserts.Backstore, trusted ass
 	return nil
 }
 
-func (run *Runner) initDeviceInfo() error {
-	const errPrefix = "cannot set device information: "
+func findTimeLowerBoundHintFile() string {
+	// uc20+
+	if osutil.FileExists(dirs.SnapModeenvFile) {
```

Comment:
it's true, but is a bit unclear at first run we'll get a good clock or not in general at the point modeenv is written, hopefully yes, but hard to know. I think it would be safer to take the highest time among modeenv and everyone of systems/*/model from seed, though it's a bit more complicated

Reply:
Thanks, I added code to look at systems/*/model now too.